components/service/glean/src/test/java/mozilla/components/service/glean/GleanTest.kt

```diff
@@ -0,0 +1,172 @@
+
+package mozilla.components.service.glean
+
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Test
+
+import org.json.JSONObject
+
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.net.HttpPingUploader
+import mozilla.components.service.glean.storages.EventsStorageEngine
+import mozilla.components.service.glean.storages.StringsStorageEngine
+
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class GleanTest {
+    @Test
+    fun `disabling metrics should record nothing`() {
+        val stringMetric = StringMetricType(
+                disabled = false,
+                category = "telemetry",
+                lifetime = Lifetime.Application,
+                name = "string_metric",
+                sendInPings = listOf("store1")
+        )
+        Glean.setMetricsEnabled(false)
+        try {
+            assertEquals(false, Glean.getMetricsEnabled())
+            stringMetric.set("foo")
+            assertNull(
+                    "Metrics should not be recorded if glean is disabled",
+                    StringsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+            )
+        } finally {
+            Glean.setMetricsEnabled(true)
+        }
+    }
+
+    @Test
+    fun `disabling event metrics should record only when enabled`() {
+        val eventMetric = EventMetricType(
+                disabled = false,
+                category = "ui",
+                lifetime = Lifetime.Application,
+                name = "event_metric",
+                sendInPings = listOf("store1"),
+                objects = listOf("buttonA")
+        )
+        Glean.setMetricsEnabled(true)
+        try {
+            assertEquals(true, Glean.getMetricsEnabled())
+            eventMetric.record("buttonA", "event1")
+            val snapshot1 = EventsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+            assertEquals(1, snapshot1!!.size)
+            Glean.setMetricsEnabled(false)
+            assertEquals(false, Glean.getMetricsEnabled())
+            eventMetric.record("buttonA", "event2")
+            val snapshot2 = EventsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+            assertEquals(1, snapshot2!!.size)
+            Glean.setMetricsEnabled(true)
+            eventMetric.record("buttonA", "event3")
+            val snapshot3 = EventsStorageEngine.getSnapshot(storeName = "store1", clearStore = false)
+            assertEquals(2, snapshot3!!.size)
+        } finally {
+            Glean.setMetricsEnabled(true)
+        }
+    }
+
+    @Test
+    fun `test sending of default pings`() {
```

Comment:
This test is a lot smaller now because it's not sending the event ping.  But I think it's still worth testing the path here (as well as more directly) because it's the only place where all the pieces involved that get integrated and you can confirm that the UUID in the path matches the UUID in the ping payload.